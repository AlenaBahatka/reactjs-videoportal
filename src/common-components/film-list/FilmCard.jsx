import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router-dom';

FilmCard.propTypes = {
	title: PropTypes.string,
	director: PropTypes.string,
	year: PropTypes.number
};

function FilmCard(props) {
	let { title, director, year, coverPicture, filmId } = props;
	const uri = `/film/${filmId}`;

	return (
		<div className="col-sm-4">
			<img src={coverPicture} alt={title} />
			<Link to={uri}>
				<h2>{title}</h2>
			</Link>
			<span> {director} </span>
			<span> {year} </span>
		</div>
	);
}

export default withRouter(FilmCard);
