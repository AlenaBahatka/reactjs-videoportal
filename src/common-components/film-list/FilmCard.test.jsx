import React from 'react';
import { shallow } from 'enzyme';
import FilmCard from './FilmCard';

describe('FilmCard', () => {
	it('should correctly render component', () => {
		let filmCardComponent = shallow(
			<FilmCard title="Some title" director="One director" year={2019} coverPicture="some picture url" />
		);
		expect(filmCardComponent).toMatchSnapshot();
	});
});
