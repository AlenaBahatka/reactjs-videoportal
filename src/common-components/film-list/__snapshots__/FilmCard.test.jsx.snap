// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard should correctly render component 1`] = `
<div
  className="col-sm-4"
>
  <img
    alt="Some title"
    src="some picture url"
  />
  <h2>
     
    Some title
     
  </h2>
  <span>
     
    One director
     
  </span>
  <span>
     
    2019
     
  </span>
</div>
`;
